skp_generator/README.md

```diff
@@ -0,0 +1,56 @@
+# SKP Generator
+
+The script in this directory generates SKPs of Flutter scenes for use
+by the Skia team in their regression and performance testing.
+
+
+## Generating the SKPs
+
+Ensure that `git` and `flutter` are available on the PATH of your
+POSIX command line.
+
+Run "build.sh" to generate the SKPs. They will all be placed in the
+`skps/` directory if everything is successful.
+
+WARNING!!! THIS SCRIPT DELETES EVERYTHING NEW IN THIS DIRECTORY.
+Only files commited to Git before running the script will survive!
+
+
+## Contributing new scenes
+
+Contributions of scenes are welcome. New scenes should be
+significantly distinct from all existing scenes, and should either
+show a scene that is expensive to rasterize, or show an interesting or
+complex scene from a real-world application.
+
+To contribute a scene, create a stateless widget in its own file in
+the `lib/tests` directory, then add a reference to that widget from
+the `lib/main.dart` file in the array near the top.
+
+Scenes may have variants, e.g. to show several frames from an
+animation. This is achieved by passing different arguments to the
+stateless widget from the array (see `lib/main.dart` for some
+examples). In general, the number of variants should be kept small.
+
+Scenes may have complex visual hierarchies but should avoid
+unnecessary widgets. For example, avoid introducing Navigators,
+inherited widgets for data models, etc. Leave only the widgets
+necessary to create the scene statically in one build.
+
+You can use https://debugger.skia.org/ to make sure that you created
+what you expected (click on EndDrawPicture at the end of the op list
+to see the final output). The output size is 800x600 at 3dpr.
+
+If you have animations, consider placing all the frames into one image
+(side by side) to simulate the cost of rendering the entire animation.
+
```

Comment:
SKPs embed image and font data. It might be fine to wait until we have this problem, but if people are submitting large numbers of scenes that have large chunks of image and font data, it may be worth replacing images with the smallest possible representation that preserves the size of the image (actual pixels don't matter for shader compilation).

For fonts we don't have a great option at this point other than to include them, see https://bugs.chromium.org/p/skia/issues/detail?id=10404

Reply:
What problem would we be solving here?

Reply:
Binary size of the SKPs.  This is probably fine to ignore until that becomes a problem, and it might not ever.

Reply:
Yeah I haven't quite figured out where the SKPs end up living right now but I'm not aware of size being an issue we need to care about yet.